pkg/machinery/resources/k8s/static_pod_list.go

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package k8s
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/talos-systems/talos/pkg/machinery/proto"
+)
+
+// StaticPodListType is type of StaticPodList resource.
+const StaticPodListType = resource.Type("StaticPodLists.kubernetes.talos.dev")
```

Comment:
I'm torn on that, but feels like we actually don't need this resource.

As soon as the request goes to the static pod HTTP server, it could list `StaticPods` and build a `v1.PodList` in memory and serve that back to the `kubelet`.